template/template.go

```diff
@@ -64,12 +64,18 @@ func init() {
 
 	HelperFuncs = template.FuncMap{
 		// Misc functions that operate on IfAddrs inputs
-		"attr":   sockaddr.IfAttr,
+		"attr":   sockaddr.IfAttrs,
 		"join":   sockaddr.JoinIfAddrs,
 		"limit":  sockaddr.LimitIfAddrs,
 		"offset": sockaddr.OffsetIfAddrs,
 		"unique": sockaddr.UniqueIfAddrsBy,
 
+		// Misc math functions that operate on a single IfAddr input
+		"math": sockaddr.IfAddrsMath,
+
+		// Misc functions that operate on IfAddr input
+		"Attr": sockaddr.IfAttr,
```

Comment:
I chewed on this a bit before settling on mixed-cases.  The problem is that `attr` accepts an `IfAddrs` as its argument, not an `IfAddr`.

The original thinking of the template functions was that sources at the start of the template pipeline would be uppercase and everything that operated on `IfAddrs` structs would be lowercase to indicate that they can't come first in the pipeline.  e.g.

https://github.com/hashicorp/go-sockaddr/blob/master/template/template.go#L60-L71

So far this is the case, but with the `math` template operation, I explicitly wanted to avoid what I did with `addr` where I did something sketchy on the grounds that it was good UX.  The behavior of `attr` is such that it only returns the attribute from the first element in `IfAddrs`.  Originally I did this because I didn't want to write `| limit 1 | attr "address"` everywhere (history note: originally the API only had `| limit 1 | join "address" ""` and that experience was awful so I conjured up `| attr "address"` to extract an individual attribute as a string).

I was already thinking about using `interface{}` but I didn't want to sacrifice type safety in the library portion of the `go-sockaddr` and this seemed like a slippery slope:

```
// IfAttrs forwards the selector to IfAttrs.Attr() for resolution.  If there the
// argument passed is IfAddrs, only the first IfAddr in the slice is used.  If
// the argument is an IfAddr, it will be used.
func IfAttrs(selectorName string, ifAddrsRaw interface{}) (string, error) {
        attrName := AttrName(strings.ToLower(selectorName))
        switch v := ifAddrsRaw.(type) {
        case IfAddrs:
                if len(v) == 0 {
                        return "", nil
                }
                attrVal, err := v[0].Attr(attrName)
                return attrVal, err
        case IfAddr:
                attrVal, err := v.Attr(attrName)
                return attrVal, err
        default:
                return "", fmt.Errorf("unable to obtain attribute %s from type %T (%v)", selectorName, ifAddrsRaw, ifAddrsRaw)
        }
}
```

... you know what.  Let me drop a facade over `sockaddr.IfAttr` and `sockaddr.IfAddrs` that does the type-unsafe `interface{}` type assertions.

I'm not sure why I didn't think of this earlier but I'm going to extend this idiom elsewhere to paper over various user issues (i.e. naked numbers in the template pipeline).  LMK what you think now.